src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -420,4 +451,88 @@ public static boolean isCorrectJavadocPosition(DetailAST blockComment) {
                 || BlockCommentPosition.isOnEnumConstant(blockComment)
                 || BlockCommentPosition.isOnAnnotationDef(blockComment);
     }
+
+    /**
+     * Checks if the supplied {@code DetailNode} argument is not a tight HTML tag.
+     * @param node the node to check
+     * @return true if the javadoc node is not a tight HTML tag; false otherwise
+     */
+    public static boolean isNonTightHtmlTag(DetailNode node) {
+        final boolean isLooseHtmlTag;
+        isLooseHtmlTag = node.getText().endsWith("TAG_START")
+                && node.getParent().getType() != getHtmlTagParentTypeWhenClosed(node);
```

Comment:
@ps-sp , as discussed with @romani - we definitely shouldn't use `Holder`

I guess using some structure to transfer `DetailNode`, flag and all additional info for violation message would be OK.